### PR TITLE
Issue #3273182 by Ressinel, dmytriikaiun, navneet0693: Fix wrong functions names of update hooks in social_group_flexible_group

### PR DESCRIPTION
**Pull request by Ressinel**
## Problem

- `social_group_flexible_update_10301` should be `social_group_flexible_group_update_10301`
- `social_group_flexible_update_10302` should be `social_group_flexible_group_update_10302`

## Solution

- To fix wrong functions names of update hooks
- Make sure that all sites that already updated to Open Social 11 will have those changes

## Issue tracker
- https://www.drupal.org/project/social/issues/3273182

## Theme issue tracker
N/A

## How to test
- [ ] Install Open Social 10.2.x
- [ ] Update to Open Social 11.x
- [ ] Changes from `social_group_flexible_update_10301` and `social_group_flexible_update_10302` won't applied since wrong functions names of update hooks
- [ ] Checkout to a branch of this PR, redo previous steps, and changes from wrong update hooks will be applied

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
Fixed wrong functions names of update hooks in social_group_flexible_group.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2852-4tg0dnkmyhkeivrvsvdchao75r38zkmo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6249adb97dc9604ba3d3fbaf

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install [hunk 1]
```diff
@@ -381,23 +381,22 @@ function social_group_flexible_group_update_8912() {
  * Allow CM+ translate "Flexible Group" group type.
  */
 function social_group_flexible_update_10301() {
-  foreach (['sitemanager', 'contentmanager'] as $role_id) {
-    user_role_grant_permissions($role_id, ['translate flexible_group group']);
-  }
+  // Removed hook due to that it doesn't apply since wrong current function
```
Comment (navneet0693) [inline modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:384]: `//s` 

Removed code in the function as it is neither an update hook nor it is called anywhere. The name of the function should have been `social_group_flexible_group_update_10301() ` which qualifies it to be an update hook. We have moved the code too a new update hook().

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install [hunk 2]
```diff
@@ -381,23 +381,22 @@ function social_group_flexible_group_update_8912() {
...
 function social_group_flexible_update_10302() {
-  /** @var \Drupal\update_helper\Updater $updateHelper */
-  $updateHelper = \Drupal::service('update_helper.updater');
-
-  // Execute configuration update definitions with logging of success.
-  $updateHelper->executeUpdate('social_group', 'social_group_flexible_group_update_10302');
-
-  // Output logged messages to related channel of update execution.
-  return $updateHelper->logger()->output();
+  // Removed hook due to that it doesn't apply since wrong current function
```
Comment (navneet0693) [inline modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:395]: Same as above, just change the update hook name.

**Review by navneet0693 — CHANGES_REQUESTED** [review]
@Ressinel Very well spotted !!!

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install (hunk 1 above)
Comment (Ressinel) [reply modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install]: Updated

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install (hunk 2 above)
Comment (Ressinel) [reply modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install]: Updated

**Comment by Ressinel** [comment]
> Very well spotted !!!

Actually, @dmytriikaiun  spotted it. So, Very well spotted @dmytriikaiun 🙂

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2852-4tg0dnkmyhkeivrvsvdchao75r38zkmo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6249adb97dc9604ba3d3fbaf

**Review by navneet0693 — APPROVED** [review]

**Comment by Ressinel** [comment]
🍒-picked to `11.0.x`, `11.1.x`, and `11.2.x`

**Pull request merged**